crates/repository/src/lib.rs

```diff
@@ -364,6 +364,15 @@ impl Repository {
     pub async fn set_location(&self, target: &str, location: &str) -> Result<()> {
         let target_location = self.parameters_root().join(format!("{target}_location"));
         let new_location = Path::new(location);
+        let new_location_path = self.parameters_root().join(location);
+        if !new_location_path.exists() {
```

Comment:
Use a https://docs.rs/tokio/latest/tokio/fs/fn.try_exists.html instead. Your code is executed in an async context, we can thus use async functions to perform io.

Reply:
Done